packages/core/src/index.ts

```diff
@@ -603,132 +617,120 @@ class Hopr extends EventEmitter {
-      currentChannels.push(channel)
-      if (await this.isAllowedAccessToNetwork(channel.destination.toPeerId())) {
-        this.networkPeers.register(channel.destination.toPeerId(), NetworkPeersOrigin.STRATEGY_EXISTING_CHANNEL) // Make sure current channels are 'interesting'
-      } else {
-        error(`Protocol error: Strategy is monitoring non-registered peer ${channel.destination.toString()}`)
+      let incomingChannels = 0
+      for (const channel of await this.getChannelsTo(selfAddr)) {
+        metric_channelBalances.set(
+          [channel.source.toAddress().toHex(), 'in'],
+          +ethersUtils.formatEther(channel.balance.toBN().toString())
+        )
+        incomingChannels++
       }
-    }
-
-    metric_inChannelCount.set(incomingChannels)
-    metric_outChannelCount.set(outgoingChannels)
-
-    // Remove last `\n`
-    verbose(out.substring(0, out.length - 1))
 
-    let balance: Balance
-    try {
-      balance = await this.getBalance()
+      metric_inChannelCount.set(incomingChannels)
+      metric_outChannelCount.set(outgoingChannels)
     } catch (e) {
-      throw new Error('failed to getBalance, aborting tick')
+      log(`error: failed to update channel metrics`, e)
     }
+  }
 
-    let allOutgoingChannels: OutgoingChannelStatus[] = Array.from(
-      currentChannels
-        .filter((c) => c.status == ChannelStatus.Open && c.source.toAddress().eq(selfAddr))
-        .map((c) => new OutgoingChannelStatus(c.destination.toPeerId().toString(), c.balance.toBN().toString()))
-    )
-
-    let peersIterator = this.networkPeers.all().map((p) => p.toString())[Symbol.iterator]
-
-    const tickResult: StrategyTickResult = this.strategy.tick(
-      balance.toBN(),
-      peersIterator(),
-      allOutgoingChannels,
-      (peer_id_str: string) => this.networkPeers.qualityOf(peerIdFromString(peer_id_str))
-    )
-
-    verbose(`strategy wants to close ${tickResult.to_close().length} channels`)
-
-    let allClosedChannels = Array.from(tickResult.to_close())
-
-    // Enhance with all channels which are in PendingToClose state to close them for good
-    allClosedChannels.push(
-      ...currentChannels
-        .filter(
-          (c) =>
-            c.status == ChannelStatus.PendingToClose && !allClosedChannels.includes(c.destination.toPeerId().toString())
-        )
-        .map((c) => c.destination.toPeerId().toString())
-    )
-
-    verbose(`strategy wants to finalize closure of ${allClosedChannels.length - tickResult.to_close().length} channels`)
+  // On the strategy interval, poll the strategy to see what channel changes
+  // need to be made.
+  private async tickChannelStrategy() {
+    verbose('strategy tick', this.status, this.strategy.name)
+    if (this.status != 'RUNNING') {
+      throw new Error('node is not RUNNING')
+    }
 
+    let tickResult: StrategyTickResult
     try {
+      // Retrieve all outgoing channels
+      const outgoingChannels = await this.getChannelsFrom(this.getEthereumAddress())
```

Comment:
Using sth like `this.getChannelsFromIterable` might not block the thread until the DB query has been finished

Reply:
`getChannelsFromIterable` does not exist, but I can create it.